### ISSUE TITLE
Set more Purchase properties on iOS

### ISSUE THREAD

**Issue by Redth** [jamesmontemagno/InAppBillingPlugin#456]
I've noticed with my auto renewing subs that there's a couple properties that aren't set properly on iOS and though they don't _really_ matter it would be nice if we they were set correctly.

1. AutoRenewing - This one is always false even though it's an auto-renewing sub.  Not sure if there's enough data in the receipt to discern this?
2. IsAcknowledged - This could always be true for subscriptions on iOS, right?

**Comment by jamesmontemagno** [comment]
1.) I don't think it is available. Since you have to assign it as auto-renewing at time of creation, you will know if it is or not. Although I agree that it is not great that is always false. I should return an enum of AutoRenewing, NonRenewing, or Unknown. https://developer.apple.com/documentation/storekit/skproduct

2.) So, yeah for this one if it is a consumable and it is returned then it has not been acknowledged, technically. For subscriptions it would always be Acknowledged. I think here.... I agree I should maybe return an enum of "IsAcknowledged", "NotAcknowledged", "Unknown".

Thoughts?

**Comment by Redth** [comment]
Enums I guess are fine but the context I want it in is to know if I need to call acknowledge or not so unknown isn't a super helpful state? Not intuitive I guess anyway. 

If iOS has no acknowledgment concept then it seems fine to return true. If that changes one day you could enum it but why make an api break now if there's no clear benefit?

**Comment by jamesmontemagno** [comment]
Yeah... my understanding is that you can always call finish all the time basically and if it is already finished no big deal. The only thing is consumables which will dissappear from your history once you finish them.
